docs/writing-stories/args.md

```diff
@@ -207,6 +207,42 @@ Similarly, special formats are available for dates and colors. Date objects will
 
 Args specified through the URL will extend and override any default values of args set on the story.
 
+## Setting args from within a story
+
+Components with interactivity often need their containing component, or page, to respond to events, modify their state, then pass a changed arg back to the rendered component to reflect the change. For example, you would want a Switch component to be checked by a user and the arg shown in Storybook to reflect the change. This can be accomplished by using the `useArgs` hook exported by `@storybook/preview-api`:
+
+```ts
+// my-component/component.stories.tsx
+
+import { StoryObj, Meta } from "@storybook/react";
+import { useArgs } from "@storybook/preview-api";
+import { Switch } from ".";
+
+const meta: Meta<typeof Switch> = {
+    title: "Inputs/Switch",
+    component: Switch
+};
+export default meta;
+
+type Story = StoryObj<typeof Switch>;
+
+export const Example = {    
+    args: {
+        isChecked: false,
+        label: "Switch Me!"
+    },
+    render: function Render(args) {      
+      const [{ isChecked }, updateArgs] = UseArgs();
+
+      function onChange() {
+          updateArgs({ isChecked: !isChecked });
+      }
+
+      return <Switch {...args} onChange={onChange} isChecked={isChecked} />;
+    }
+};
+```
```

Comment:
@jonthenerd, if you're with it, could you update your pull request and factor in the following:
- Create a TypeScript snippet file inside the react folder of the documentation
- Do the same for a JavaScript variation
- Replace the code block here with reference to the filenames as follows:

```md
<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'react/this-pr-example-snippet.js.mdx',
    'react/this-pr-example-snippet.ts.mdx',
  ]}
/>

<!-- prettier-ignore-end -->
```

- If possible, run the `yarn pretty-docs` task on the monorepo so that the files run through prettier to avoid inconsistencies, typos, etc.

Reply:
@jonniebigodes I believe this is ready now. Thanks!

Reply:
@jonniebigodes did my changes 3 weeks ago meet your expectations?

Reply:
@jonthenerd sorry for the radio silence on this. I'll take another look tomorrow morning my time and follow up with you on this. Sounds good?